kernel/reduction.ml

```diff
@@ -303,7 +303,8 @@ let unfold_ref_with_args infos tab fl v =
   | Primitive op when check_native_args op v ->
     let c = match fl with ConstKey c -> c | _ -> assert false in
     let rargs, a, nargs, v = get_native_args1 op c v in
-    Some (a, (Zupdate a::(Zprimitive(op,c,rargs,nargs)::v)))
+    let v = zupdate infos a (Zprimitive(op,c,rargs,nargs)::v) in
```

Comment:
I think you should rather export a smart constructor for `Zprimitive` that does both the update and the primitive.

Reply:
changed